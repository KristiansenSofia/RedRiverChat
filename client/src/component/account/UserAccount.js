import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';

// Import NPM-modules
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';

// Import styles. userAccountStyles for all imported components with a style attribute and CSS-file for classNames and id.
import {userAccountStyles} from "../../styles/AccountStyles";
import '../../styles/Styles.css'

/**
 *  ChatList-component. Starting page of chat.
 *
 *  @author Jimmy
 */

class UserAccount extends Component {

    constructor(props){
        super(props);

        this.state = {
            name: 'Red River', // Change when API-requests are done!
        };

    }

    render() {

        if (this.props.state.isSignedIn === false) {
            return <Redirect to="/login" />
        }

        return (
            <div className="UserAccount">
                <Typography
                    variant="headline"
                    color="textSecondary"
                    align="left"
                    style={userAccountStyles.title}
                >
                    Hej, {this.props.state.userInfo.username}!
                </Typography>
                <p className="AccountSecondTitle">Kom igång genom att skicka ett meddelande nedan.</p>
                <div className="UserAccountButtonDiv">
                    <Button variant="raised"
                            style={userAccountStyles.button}
                            component={Link}
                            to="/chats"
                    >
                        Mina chattrum
                    </Button>
                    <Button variant="raised"
                            style={userAccountStyles.button}
                            component={Link}
                            to="/friends"
                    >
                        Mina vänner
                    </Button>
                    <Button variant="raised" style={userAccountStyles.button} /*onClick={}*/>
                        Starta videosamtal
                    </Button>
                    <Button variant="raised" style={userAccountStyles.button} /*onClick={}*/>
                        Starta livesändning
                    </Button>
                </div>
            </div>

        );
    }
}

export default UserAccount;