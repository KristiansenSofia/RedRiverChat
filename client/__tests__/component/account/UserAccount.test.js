import React from 'react';
import { MemoryRouter } from 'react-router';
import UserAccount from '../../../src/component/account/UserAccount';
import renderer from 'react-test-renderer';

test('UserAccount renders without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <UserAccount state = {{ isSignedIn: true, userInfo : { username: 'username' }}}/>
    </MemoryRouter>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
