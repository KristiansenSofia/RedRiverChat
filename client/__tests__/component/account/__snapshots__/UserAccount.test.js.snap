// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccount renders without crashing 1`] = `
<div
  className="UserAccount"
>
  <h1
    className="MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-colorTextSecondary-23 MuiTypography-alignLeft-13"
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    Hej, 
    username
    !
  </h1>
  <p
    className="AccountSecondTitle"
  >
    Kom igång genom att skicka ett meddelande nedan.
  </p>
  <div
    className="UserAccountButtonDiv"
  >
    <a
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30"
      href="/chats"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "height": 140,
          "margin": 20,
          "width": 160,
        }
      }
      tabIndex="0"
    >
      <span
        className="MuiButton-label-26"
      >
        Mina chattrum
      </span>
      <span
        className="MuiTouchRipple-root-43"
      />
    </a>
    <a
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30"
      href="/friends"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "height": 140,
          "margin": 20,
          "width": 160,
        }
      }
      tabIndex="0"
    >
      <span
        className="MuiButton-label-26"
      >
        Mina vänner
      </span>
      <span
        className="MuiTouchRipple-root-43"
      />
    </a>
    <button
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "height": 140,
          "margin": 20,
          "width": 160,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-26"
      >
        Starta videosamtal
      </span>
      <span
        className="MuiTouchRipple-root-43"
      />
    </button>
    <button
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "height": 140,
          "margin": 20,
          "width": 160,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-26"
      >
        Starta livesändning
      </span>
      <span
        className="MuiTouchRipple-root-43"
      />
    </button>
  </div>
</div>
`;
